### PR TITLE
Filter Customer/Order/Sale/Product/ProductVariant by datetime of last modification

### PR DESCRIPTION
**Pull request by rafalp**
This PR adds `updated_at` column to following models:

- `User`
- `Order`
- `Sale`
- `Product`
- `ProductVariant`

It also adds those columns to GraphQL schema as `updatedAt: DateTime!`.

Lastly, it enables filtering GraphQL lists by those columns.

# Impact

* [X] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [x] Database migration files are up to date
* [x] The changes are tested
* [x] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -48,6 +48,9 @@ All notable, unreleased changes to this project will be documented in this file.
 - Change metadata mutations to use token for order and checkout as identifier - #8426 by @IKarbowiak
   - After changes, using the order `id` for changing order metadata is deprecated
 - Add `withChoices` flag for Attribute type - #7733 by @dexon44
+`CheckoutAddPromoCode`, `CheckoutPaymentCreate` will raise a ValidationError when product in the checkout is
+unavailable - #8978 by @IKarbowiak
```
Comment (IKarbowiak) [inline CHANGELOG.md:52]: Should be removed.

**Review by IKarbowiak — APPROVED** [review]

**Review by korycins — APPROVED** [review]

File: saleor/product/migrations/0160_populate_product_datetimes.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+from django.db import migrations
+from django.db.models import F
+from django.utils import timezone
+
+
+def populate_products_datetimes(apps, _schema_editor):
+    Product = apps.get_model("product", "Product")
+
+    Product.objects.filter(updated_at__isnull=False).update(created=F("updated_at"))
+    Product.objects.filter(created__isnull=True).update(created=F("updated_at"))
```
Comment (fowczarek) [inline saleor/product/migrations/0160_populate_product_datetimes.py:12]: Shouldn't be like that. Your solution move `null` from `updated_at` field to `created` field.
```suggestion
    Product.objects.filter(created__isnull=True).update(updated_at=timezone.now())
```

File: saleor/product/migrations/0162_auto_20220209_1517.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 3.2.12 on 2022-02-09 15:17
```
Comment (fowczarek) [inline saleor/product/migrations/0162_auto_20220209_1517.py:1]: Could we merge this migration with `0159_auto_20220209_1501.py` both just adding fields?

File: saleor/product/migrations/0163_populate_product_variant_datetimes.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Generated by Django 3.2.12 on 2022-02-09 15:17
```
Comment (fowczarek) [inline saleor/product/migrations/0163_populate_product_variant_datetimes.py:1]: Could we merge this migration with `0160_populate_product_datetimes.py `?

File: saleor/product/migrations/0164_auto_20220209_1522.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 3.2.12 on 2022-02-09 15:22
```
Comment (fowczarek) [inline saleor/product/migrations/0164_auto_20220209_1522.py:1]: Could we merge this migration with `0161_auto_20220209_1511.py`?

File: saleor/product/migrations/0160_populate_product_datetimes.py (hunk 2 above)
Comment (rafalp) [reply saleor/product/migrations/0160_populate_product_datetimes.py:12]: I'll change this to `Product.objects.filter(created__isnull=True).update(created=timezone.now())`

**Review by fowczarek — APPROVED** [review]

File: saleor/order/migrations/0125_populate_order_updated_at.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# Generated by Django 3.2.12 on 2022-02-09 13:58
+
+from django.db import migrations
+
+
+def populate_orders_updated_at_date_from_events(apps, _schema_editor):
+    Order = apps.get_model("order", "Order")
+
+    for order in Order.objects.filter(updated_at__isnull=True).iterator():
+        last_event = order.events.last()
```
Comment (maarcingebala) [inline saleor/order/migrations/0125_populate_order_updated_at.py:10]: This can be a performance issue as it's making a query in for-loop.

**Review by maarcingebala — CHANGES_REQUESTED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**